### PR TITLE
newest version of Fabric.js 2.3.3

### PR DESCRIPTION
**Pull request by Mazen-Embaby**
-fix feature selecting all objects and sending back and forward respect to others
-change all function that removed in the new Fabricjs v 2.3.3 like:
     setActiveGroup(group)
     getActiveGroup();
     deactivateAll();
     discardActiveGroup();
     deactivateAllWithDispatch();
     obj.Shadow()

### CODE REVIEW

**Comment by danielktaylor** [comment]
Thanks @Mazen-Embaby - I will review in the next couple of days

File: src/js/app/fabricUtils.js [hunk 1]
```diff
@@ -9,161 +9,115 @@ Good references:
...
-      canvas.bringForward(obj);
-    }
-  }
-}
+
+// TODO Fabric.js might do this for us now that we've on version >1.5
 
 // This is the shadow-less version
 /*
 function getImageBounds() {
```
Comment (danielktaylor) [inline src/js/app/fabricUtils.js]: I think that this function is dead code, and can be removed.

**Comment by danielktaylor** [comment]
Hi @Mazen-Embaby, thank you for working on upgrading this to use the newest version of Fabric.js.

A couple of small requests before merging:

* Please remove any dead code, rather than commenting it out
* Remove any debugging statements added using console.log(...)
* Fix up the indentation to match the project
* I tested a bit, and there are a few places that errors get thrown:

screen shot 2018-07-04 at 5 05 17 pm
screen shot 2018-07-04 at 5 05 23 pm
screen shot 2018-07-04 at 5 05 33 pm

I can do another code review & QA test after updates. Thanks again.

**Comment by Mazen-Embaby** [comment]
Thanks @danielktaylor,  I will work on in the next days.

File: src/js/app/fabricUtils.js (hunk 1 above)
Comment (Mazen-Embaby) [reply src/js/app/fabricUtils.js]: You're are using it in the export .

**Comment by Mazen-Embaby** [comment]
Hello @danielktaylor  I am glad to hear from you and I appreciate your effort and work in this project, 
-First bug: Not solved and it's in the old version.
Is related to security, doing HttpRequest to a different domain than your page is on. So the browser is blocking it as it usually allows a request in the same origin for security reasons. We need to do something different when you want to do a cross-domain request. using cross .
It’s written in the FabricJS site as follow 
“Note that SVG is fetched via XMLHttpRequest, so it needs to conform to SOP (Same Origin Policy)”

-Second bug: It's exist in the old version

-Third bug:It's fixed 

I take care of other things .

**Pull request merged**